lib/sqlalchemy/dialects/sqlite/base.py

```diff
@@ -1853,6 +1853,7 @@ def get_pk_constraint(self, connection, table_name, schema=None, **kw):
             constraint_name = result.group(1) if result else None
 
         cols = self.get_columns(connection, table_name, schema, **kw)
+        cols.sort(key=lambda col: col.get("primary_key"))
```

Comment:
**mike bayer** (zzzeek) wrote:

so we identified / fixed the same issue in sqlite?  amazing!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2320

Reply:
Why is this line necessary? It appears to be sorting the cols list by the bool `primary_key`, however the code later just extracts elements from `cols` where `col[primary_key] == True`. The ordering of those elements should be unaffected by this `sort`.

```
        for col in cols:
            if col["primary_key"]:
                pkeys.append(col["name"])
```

I hope you don't mind the question from the peanut gallery. I was linked to this PR from the [1.3.21 Release Notes](https://docs.sqlalchemy.org/en/14/changelog/changelog_13.html#change-1.3.21).

Reply:
Hi @AveshCSingh, `col["primary_key"]` is an integer value that designates the column's order within the primary key constraint. I added the sort to ensure that the columns would be added to `pkeys` in the same order that they were defined in the constraint.

Reply:
Thank you for explaining this @fulpm. To fresh eyes it looked like a boolean, but I see now that it is the column order within the primary key constraint.